src/value_generators/include/value_generators/DocumentGenerator.hpp

```diff
@@ -36,11 +36,17 @@ class InvalidValueGeneratorSyntax : public std::invalid_argument {
     using std::invalid_argument::invalid_argument;
 };
 
+struct GeneratorArgs {
```

Comment:
@rtimmons this was the logical conclusion of this, with it included here for the third constructor below, but it can't really be passed into the other two. Let me know what you think. If you prefer this, we go with it. If not, I'll back it out.